ctk/common/src/test/kotlin/org/codice/compilance/verification/core/response/CoreLogoutResponseProtocolVerifierSpec.kt

```diff
@@ -0,0 +1,94 @@
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.compilance.verification.core.response
+
+import io.kotlintest.matchers.string.shouldContain
+import io.kotlintest.shouldThrow
+import io.kotlintest.specs.StringSpec
+import org.codice.compliance.Common
+import org.codice.compliance.SAMLComplianceException
+import org.codice.compliance.SAMLCore_3_7_3_2_b
+import org.codice.compliance.SAMLCore_3_7_3_2_d
+import org.codice.compliance.utils.PERSISTENT_ID
+import org.codice.compliance.utils.RESPONDER
+import org.codice.compliance.utils.SUCCESS
+import org.codice.compliance.utils.TestCommon.Companion.REQUEST_ID
+import org.codice.compliance.utils.TestCommon.Companion.currentSPIssuer
+import org.codice.compliance.verification.core.responses.CoreLogoutResponseProtocolVerifier
+import org.codice.security.saml.SamlProtocol.Binding.HTTP_POST
+import org.joda.time.DateTime
+import org.opensaml.saml.common.SAMLVersion.VERSION_20
+import org.opensaml.saml.saml2.core.impl.IssuerBuilder
+import org.opensaml.saml.saml2.core.impl.LogoutRequestBuilder
+import org.opensaml.saml.saml2.core.impl.NameIDBuilder
+import java.time.Instant
+import java.util.UUID
+
+class CoreLogoutResponseProtocolVerifierSpec : StringSpec() {
+
+    private val logoutRequest by lazy {
+        REQUEST_ID = "a" + UUID.randomUUID().toString()
+
+        LogoutRequestBuilder().buildObject().apply {
+            issuer = IssuerBuilder().buildObject().apply { value = currentSPIssuer }
+            id = REQUEST_ID
+            version = VERSION_20
+            issueInstant = DateTime()
+            destination = "https://localhost:8993/services/idp/login"
+            nameID = NameIDBuilder().buildObject().apply {
+                nameQualifier = "https://localhost:8993/services/idp/login"
+                spNameQualifier = currentSPIssuer
+                format = PERSISTENT_ID
+                value = "admin"
+            }
+        }
+    }
+
+    init {
+        "logout response with correct second-level status code should pass" {
+            Common.buildDom(createLogoutResponse(SUCCESS)).let {
+                CoreLogoutResponseProtocolVerifier(logoutRequest, it, HTTP_POST, SUCCESS).verify()
+            }
+        }
+
+        "logout response with incorrect second-level status code should fail" {
+            Common.buildDom(createLogoutResponse(RESPONDER)).let {
+                shouldThrow<SAMLComplianceException> {
+                    CoreLogoutResponseProtocolVerifier(logoutRequest, it, HTTP_POST, SUCCESS)
+                        .verify()
+                }.apply {
+                    this.message?.shouldContain(SAMLCore_3_7_3_2_b.message)
+                    this.message?.shouldContain(SAMLCore_3_7_3_2_d.message)
+                }
+            }
+        }
```

Comment:
❓ Should we also have a test for:
1. when there is no secondary status code (i.e. `null`) and we're expecting one, it will throw an exception (this could utilize the second test's code, just with different input)
2. When we're not expecting a status code, it can have any secondary status code and it won't throw an exception (I see some benefit in doing an array of a handful of values here to make sure none of them throw an exception)